omegaconf/basecontainer.py

```diff
@@ -232,7 +232,16 @@ def convert(val: Node) -> Any:
                     )
                 else:
                     retdict[key] = convert(node)
-            return retdict
+
+            if structured_config_mode == SCMode.INSTANTIATE and is_structured_config(
+                conf._metadata.object_type
+            ):
+                retstruct = _instantiate_structured_config_impl(
```

Comment:
I'd simplify by changing retdict to ret and type is as Any.

Reply:
Done in https://github.com/omry/omegaconf/pull/502/commits/d6e9749cacf127bcd3196854b6c9945014537cbd.